### PR TITLE
feat: Add flux tool.

### PR DESCRIPTION
**Pull request by danports**
Needed for https://github.com/renovatebot/renovate/pull/13675.

Haven't tried testing this yet.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -78,6 +78,7 @@ jobs:
             golang,
             java,
             jb,
+            flux,
```
Comment (viceice) [inline .github/workflows/build.yaml:81]:
```suggestion
```

simply add to `latest/Dockerfile` and bionic test

File: src/usr/local/buildpack/tools/flux.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+tool_path=$(find_tool_path)
+if [[ -z "${tool_path}" ]]; then
+  if [ -z $TOOL_VERSION ]; then
+    TOOL_VERSION=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest -sL | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/' | cut -c 2-)
+  fi
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:7]:
```suggestion
```

I don't think we need this for renovate

File: src/usr/local/buildpack/tools/flux.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  INSTALL_DIR=$(get_install_dir)
+  base_path=${INSTALL_DIR}/${TOOL_NAME}
+  tool_path=${base_path}/${TOOL_VERSION}
+  mkdir -p ${tool_path}
+
+  ARCH=amd64
+  BIN_URL="https://github.com/fluxcd/flux2/releases/download/v${TOOL_VERSION}/flux_${TOOL_VERSION}_linux_${ARCH}.tar.gz"
+  curl -sL ${BIN_URL} -o /tmp/flux.tar.gz
+  mkdir -p /tmp/flux
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:16]:
```suggestion
```

File: src/usr/local/buildpack/tools/flux.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  tar -C /tmp/flux/ -zxvf /tmp/flux.tar.gz
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:17]: You should have write rights to `${tool_path}`
```suggestion
  tar -C ${tool_path} -zxvf /tmp/flux.tar.gz
```

is there only the `flux` cli inside the gzip file? if not, can we only extract the cli?

File: src/usr/local/buildpack/tools/flux.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  require_root
+  cp /tmp/flux/flux ${tool_path}
+
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:21]:
```suggestion
```

File: src/usr/local/buildpack/tools/flux.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  rm -rf /tmp/flux/ /tmp/flux.tar.gz
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:22]:
```suggestion
```

`/tmp` is automatically cleaned up

File: src/usr/local/buildpack/tools/flux.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+link_wrapper ${TOOL_NAME} ${tool_path}/bin
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:25]:
```suggestion
link_wrapper ${TOOL_NAME} ${tool_path}
```
 I don't think it has a bin subfolder 😉

File: src/usr/local/buildpack/tools/flux.sh [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+flux -v
```
Comment (viceice) [inline src/usr/local/buildpack/tools/flux.sh:27]:
```suggestion
flux -v

```

final newline

**Review by viceice — CHANGES_REQUESTED** [review]

File: src/usr/local/buildpack/tools/flux.sh (hunk 4 above)
Comment (danports) [reply src/usr/local/buildpack/tools/flux.sh:17]: Yep, there's just a single binary in the `tar.gz` file.

**Review by viceice — APPROVED** [review]

**Pull request merged**